qiskit/aqua/operators/operator_base.py

```diff
@@ -0,0 +1,272 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Base Operator """
+
+from abc import ABC, abstractmethod
+import numpy as np
+
+from qiskit.circuit import ParameterExpression, ParameterVector
+
+from qiskit.aqua import AquaError
+
+
+class OperatorBase(ABC):
+    """ An square binary Operator can be defined in a two equivalent ways:
+        1) A functional, taking a complex function over a binary alphabet
+           of values to another binary function
+        2) A complex function over two values of a binary alphabet
+
+    """
+
+    @property
+    def name(self):
```

Comment:
Missing type hints for parameters and return types on this and a number of others in this module

Reply:
Added